src/vs/workbench/contrib/terminal/browser/terminalGroup.ts

```diff
@@ -409,22 +409,20 @@ export class TerminalGroup extends Disposable implements ITerminalGroup {
 	}
 
 	moveInstance(instances: ITerminalInstance | ITerminalInstance[], index: number, position: 'before' | 'after'): void {
-		const instances2 = Array.isArray(instances) ? instances : [instances];
-		for (const instance of instances2) {
-			const index = this.terminalInstances.indexOf(instance);
-			if (index === -1) {
-				return;
-			}
+		const instancesArray = Array.isArray(instances) ? instances : [instances];
+		const hasInvalidInstance = instancesArray.some(instance => !this.terminalInstances.includes(instance));
+		if (hasInvalidInstance) {
```

Comment:
shouldn't this be `if (!hasValidInstance)`?

Reply:
I think the `hasInvalidInstance ` was trying to indicate the source array has at least one instance not found in `this. terminalInstances`. But I guess `!hasValidInstance` and `hasInvalidInstance` are kind of the same?

Reply:
oh, I misread this. 👍🏼